examples/misc/lib/effective_dart/usage_bad.dart

```diff
@@ -76,6 +76,26 @@ void miscDeclAnalyzedButNotTested() {
     // #enddocregion use-tear-off
   };
 
+  {
+    // #docregion default-separator
+    void insert(Object item, {int at: 0}) {
+      // ...
+    }
+    // #enddocregion default-separator
+  }
+
+  {
+    // #docregion default-value-null
+    void error([String message = null]) {
+      if (message != null) {
+        stderr.write(message);
+      } else {
+        stderr.write("\n");
+      }
+    }
+    // #enddocregion default-value-null
```

Comment:
Since the point of this example concerns parameters, how about making the body more compact?

```dart
void error([String message = null]) {
  stderr.write(message ?? '\n');
}
```

Also note the use of single quotes. Or, even more compact would be:

```dart
    void error([String message = null]) => //!<br>
        stderr.write(message ?? '\n');
```

(The special marker `//!<br>` gets removed. Its there to force dartfmt to insert a line break.)

Reply:
Great idea. Done.